src/Microsoft.OData.Core/ODataMessageWriterSettings.cs

```diff
@@ -181,6 +182,13 @@ public ODataUri ODataUri
         /// </summary>
         public ODataMetadataSelector MetadataSelector { get; set; }
 
+        /// <summary>
+        /// Gets or sets the new line character sequence used when writing multipart messages
+        /// see https://tools.ietf.org/html/rfc2046#section-5.1.1
+        /// A TextWriter uses OS specific newline but rfc2046 requires it to be CRLF.
+        /// </summary>
+        public string MultipartNewLine { get; set; }

```

Comment:
>MultipartNewLine [](start = 22, length = 16)

maybe just "NewLine"

Reply:
By the way, it's only for the law value writer, what about other writer, for the JsonWriter? 

---
In reply to: [463717992](https://github.com/OData/odata.net/pull/1844#discussion_r463717992) [](ancestors = 463717992)

Reply:
> > MultipartNewLine [](start = 22, length = 16)
> 
> maybe just "NewLine"

This is specifically a setting for ODataBatchWriter and how it writes multipart messages, I would prefer to qualify it that way.

Reply:
> By the way, it's only for the law value writer, what about other writer, for the JsonWriter?
> 
> In reply to: [463717992](https://github.com/OData/odata.net/pull/1844#discussion_r463717992) [](ancestors = 463717992)

Json Writer is a wrapper arround the same TextWriter

line 177 RawValueWriter
```
    this.textWriter = new StreamWriter(nonDisposingStream, this.encoding) { NewLine = this.settings.MultipartNewLine };
    this.jsonWriter = new JsonWriter(this.textWriter, isIeee754Compatible: false);
```